Mathlib/Algebra/Hom/Centroid.lean

```diff
@@ -514,4 +515,46 @@ def commRing (h : ∀ a b : α, (∀ r : α, a * r * b = 0) → a = 0 ∨ b = 0)
 
 end NonUnitalRing
 
+
+section StarRing
+
+variable [NonUnitalNonAssocSemiring α] [StarRing α]
+
+instance : Star (CentroidHom α) where
+  star f :=
+  { toFun := fun a => star (f (star a))
```

Comment:
Can you add a docstring that states that `star f a = star (f (star a))`, and gives a reference for this definition? I've seen a handful of cases where there's no obvious canonical star structure, which a reference would allay any concerns over.

Reply:
@eric-wieser I've been thinking about this. I haven't been able to find a precise reference, but from my perspective, this is a consequence of a *-ring being a *-triple. The definition of a centroid homomorphism `T` on a *-triple requires the existence of another map `T^*`, and a mild condition can be placed on the *-triple guaranteeing uniqueness of `T^*`. In the case of a *-triple generated from a *-ring, the algebraic centroid is a *-subalgebra of the triple centroid. A further mild condition makes them coincide. The involution I've defined here coincides with the triple centroid one when the uniqueness condition is in force.

Perhaps a better approach than this PR would be for me to develop the theory outlined above in Mathlib and then treat this as a special case?

Reply:
Furthermore, this definition makes the natural (iso|homo)morphism defined in #7843 into a *-(iso|homo)morphism.

Reply:
Can you add that  (iso|homo)morphism result to this PR?